src/reanimated2/initializers.ts

```diff
@@ -118,11 +116,6 @@ function setupRequestAnimationFrame() {
       // is added and then use it to execute all the enqueued callbacks. Once
       // the callbacks are run, we clear the array.
       nativeRequestAnimationFrame((timestamp) => {
-        if (lastNativeAnimationFrameTimestamp >= timestamp) {
```

Comment:
I think it'd be safer to keep this check here. I think we don't want to guarantee that we will always call nativeRequestAnimationFrame only once in a given frame. The check on the native site doesn't give us that guarantee either, because if we enqueue it twice in one frame, there will be two callbacks in `mFrameCallbacks` list and will be triggered twice. Also, this check will be relevant to iOS because we also don't have that guarantee there.


The essence of this issue wasn't that we would get the callback triggered twice, but that callback scheduled with nativeRequestAnimationFrame in a given frame would be triggered with the same frame's timestamp. We still want to prevent executing this callback more than once in a given frame

Reply:
Resolved in https://github.com/software-mansion/react-native-reanimated/pull/4579/commits/579335b94c7b3773cf56c58d62175938b40bde5e